flight/PiOS/posix/pios_delay.c

```diff
@@ -33,17 +33,80 @@
 #include "pios.h"
 #include "time.h"
 
+#include <time.h>
+
+/**
+ * This is the value used as a base.  Strictly not required, as times
+ * can be expected to wrap... But it makes sense to get sane numbers at
+ * first that will agree with the PIOS_Thread ones etc.
+ */
+static uint32_t base_time;
+
+#ifdef __MACH__
+#include <mach/mach_time.h>
+
+typedef int clockid_t;
+
+#ifndef CLOCK_MONOTONIC
+#define CLOCK_MONOTONIC 1
+#endif
+
+//clock_gettime is not implemented on OSX
+int clock_gettime(int clk_id, struct timespec* t) {
+	(void) clk_id;
+
+	uint64_t tm = mach_absolute_time();
+	static int numer = 0, denom = 0;
+
+	if (!numer) {
+		mach_timebase_info_data_t tb;
+
+		kern_return_t ret = mach_timebase_info(&tb);
+
+		if (ret != KERN_SUCCESS) abort();
+
+		numer = tb.numer;
+		denom = tb.denom;
+	}
+
+	tm *= numer;
+	tm /= denom;
+
+	t->tv_nsec = tm % 1000000000;
+	t->tv_sec  = tm / 1000000000;
+
+	return 0;
+}
+#endif
```

Comment:
Can you comment these so they can be easily matched up to the corresponding if please?